lib/reply.js

```diff
@@ -169,6 +156,36 @@ function wrapReplySend (reply, payload) {
   }
 }
 
+function handleError (reply, err) {
+  if (!reply.res.statusCode || reply.res.statusCode < 400) {
+    reply.res.statusCode = 500
+  }
+
+  reply._req.log.error({ res: reply.res, err }, err.message)
+
+  const store = reply.store
+  const errorHandler = store && store.errorHandler
+
+  if (errorHandler && !reply._errored) {
+    reply.sent = false
+    reply._errored = true
+    errorHandler(err, reply)
+    return
+  }
+
+  reply.res.setHeader('Content-Type', 'application/json')
+  setImmediate(
+    wrapReplyEnd,
+    reply,
+    stringify(xtend({
+      error: statusCodes[reply.res.statusCode + ''],
+      message: err.message,
+      statusCode: reply.res.statusCode
+    }, reply._extendServerError && reply._extendServerError(err)))
```

Comment:
If `reply._extendServerError` is not a function, doest `xtend` work correctly? Is there a test for that?

Looking at [here](https://github.com/Raynos/xtend/blob/master/immutable.js#L8) there're some troubles

Reply:
We already discussed it here https://github.com/fastify/fastify/pull/236 It's up to the developer to use the interface correctly. It's also documented.

Reply:
@delvedor I have another point. We should remove the dependency to `xtend` because the performance isn't better than `Object.assign`. `Object.assign` is almost 100% faster.

```
benchExtend*1000: 0.520ms
benchObjectAssign*1000: 0.249ms
benchExtend*1000: 0.619ms
benchObjectAssign*1000: 0.330ms
``` 
https://github.com/Raynos/xtend/issues/32#issuecomment-307484471
https://github.com/styfle/xtend-es6 :D

Reply:
Is this confirmed in Node 6 as well?

Reply:
Verified on 6.0.0

```
benchExtend*1000: 0.534ms
benchObjectAssign*1000: 0.199ms
benchExtend*1000: 0.481ms
benchObjectAssign*1000: 0.238ms
```

Reply:
@StarpTech I mean: if there's the check, the `reply._extendServerError` can be a falsy value. In that case `xtend` fails.

So, why is there this check?

Reply:
We need the check because there isn't a default handler. With `Object.assign` we fixed your case.